### ISSUE TITLE
[no-method-signature] desconsider `Readonly<{TSMethodSignature}>`

### ISSUE THREAD

**Issue by rafaelss95** [jonaskello/eslint-plugin-functional#261]
# Bug Report
According to [**no-method-signature docs**](https://github.com/jonaskello/eslint-plugin-functional/blob/master/docs/rules/no-method-signature.md), we shouldn't use `TSMethodSignature` because it'll be mutable, however we can enforce immutability even for `TSMethodSignature` by using `Readonly<{...}>`.

## Expected behavior
I'd expect the rule to not flag these cases.

## Actual behavior
Lint warning:
<img width="667" alt="Captura de Tela 2021-08-22 às 21 20 56" src="https://user-images.githubusercontent.com/11965907/130374977-b2382a82-4bac-498e-a807-f2e2ac4f73b3.png">

## Steps to reproduce
```ts
type FileUploaderProps = Readonly<{
  onSubmit(file: File): void;
}>;
```

or (ok, this is ugly, but works 😅):

```ts
interface FileUploaderProps extends Readonly<{
  onSubmit(file: File): void;
}>{};
```

## Proposed changes
Instead of just checking for `TSMethodSignature`, ensure there's no parent `Readonly <{...}>`.

**Comment by rafaelss95** [comment]
Can I work on it, @RebeccaStevens?

**Comment by RebeccaStevens** [comment]
Yes PRs are always open.

The way I think this should be implemented is by adding a new option to the rule called `ignoreIfReadonly` which is `true` by default.

If the change is quite small (which I imagine it will be) we should be able to release it straight away.
If it's more complicated or wants to use some more advanced features (such as `isReadonly`) like the future [`prefer-readonly-type-declaration` rule](#259) is doing, we'll probably release it in the next major release.
